libs/utils/trace.py

```diff
@@ -136,10 +136,14 @@ def __init__(self, platform, data_dir, events,
 
         # If we don't know the number of CPUs, check the trace for the
         # highest-numbered CPU that traced an event.
-        if 'cpus_count' not in self.platform:
+        if self.platform and 'cpus_count' not in self.platform:
```

Comment:
I'm not sure to see why we need to check that self.platform is not empty before checking for the key. Can it be something else than a dict (e.g. None) ? If so, the following additions would crash since it is adding a key to self.platform (which won't work if self.platform is None).

Reply:
mmm... right... so... I think the issue is the ambiguity of the platform parameter.
If a dummy user (like myself) passes in something which is not None or an iterable, we end up with a failure in the line above.